### PR TITLE
installation for arm64 

### PR DESCRIPTION
**Pull request by tmsincomb**
pyproject will auto detect if arm is being used and will install neon branch if found.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -58,7 +61,6 @@ numpy = "^1.24.2"
 [tool.poetry.group.dev.dependencies]
 pytest = ">=7.1.0"
 flake8 = "4.0.1"
-pylint = "2.12.2"
```
Comment (tmsincomb) [inline pyproject.toml:61]: pylint was holding python 3.11 back in poetry: wrapt 1.13 deprecated and will fail with pylint 2.12

File: pyproject.toml [hunk 2]
```diff
@@ -48,7 +48,10 @@ Levenshtein = "^0.20.5"
 pyarrow = ">=6.0.1"
 semantic-version = "^2.10.0"
 PyYAML = "^6.0"
-pyhmmer = "^0.7.1"
+pyhmmer = [
+    { version = "^0.7.4", markers= "sys_platform == 'darwin' and platform_machine != 'arm64'"  },
+    { git = "https://github.com/althonos/pyhmmer", branch = "neon-support", markers= "platform_machine == 'arm64'" }
```
Comment (tmsincomb) [inline pyproject.toml:53]: neon branch currently at 7.4

**Review by jwillis0720 — APPROVED** [review]

File: pyproject.toml [hunk 3]
```diff
@@ -37,9 +37,9 @@ issues = "https://github.com/jwillis0720/sadie/issues"
 sadie = "sadie.app:sadie"
 
 [tool.poetry.dependencies]
-python = ">=3.8,<3.11"
+python = ">=3.8,<3.12"
```
Comment (tmsincomb) [inline pyproject.toml:40]: biopython updated so this should pass

File: pyproject.toml [hunk 4]
```diff
@@ -37,9 +37,9 @@ issues = "https://github.com/jwillis0720/sadie/issues"
...
 biopython = "1.80"
-click = ">=7.0"
+click = ">=8.0"
```
Comment (tmsincomb) [inline pyproject.toml:42]: natural bump without conflict

**Pull request merged**